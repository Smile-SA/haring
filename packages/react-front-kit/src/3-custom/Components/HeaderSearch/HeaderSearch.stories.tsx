import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '../../../hooks/useStorybookArgsConnect';

import { HeaderSearch as Cmp } from './HeaderSearch';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onChange: 'value',
      });
      return (
        <div style={{ position: 'relative' }}>
          <Story args={{ ...args }} />
        </div>
      );
    },
  ],
  tags: ['autodocs'],
  title: '3-custom/Components/HeaderSearch',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const HeaderSearch: IStory = {
  args: {
    clearButtonAriaLabel: 'Clear',
    opened: false,
    value: '',
  },
};
