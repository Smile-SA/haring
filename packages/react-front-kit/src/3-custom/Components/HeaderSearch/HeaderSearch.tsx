import type { BoxProps } from '@mantine/core';
import type {
  ChangeEventHandler,
  FormEvent,
  MouseEvent,
  MouseEventHandler,
} from 'react';

import { Box, CloseButton, Input, createStyles } from '@mantine/core';
import { useEffect, useRef } from 'react';

const useStyles = createStyles((theme) => ({
  search: {
    background: theme.colorScheme === 'dark' ? theme.black : theme.white,
    bottom: 0,
    left: 0,
    padding: '30px 64px',
    position: 'absolute',
    right: 0,
    translate: '0 100%',
  },
}));

interface IHeaderSearchProps extends BoxProps {
  clearButtonAriaLabel?: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onClear?: MouseEventHandler<HTMLButtonElement>;
  onSubmit?: (event: FormEvent) => void;
  opened?: boolean;
  value?: string;
}

export function HeaderSearch(props: IHeaderSearchProps): JSX.Element {
  const {
    clearButtonAriaLabel = 'Clear',
    opened,
    onChange,
    onClear,
    onSubmit,
    value,
    ...boxProps
  } = props;
  const { classes } = useStyles();
  const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    if (inputRef.current && opened) {
      inputRef.current.focus();
    }
  }, [opened]);

  function handleSearchClear(event: MouseEvent<HTMLButtonElement>): void {
    onClear?.(event);
    if (inputRef.current && opened) {
      inputRef.current.focus();
    }
  }

  return (
    <Box className={classes.search} {...boxProps}>
      <form onSubmit={onSubmit}>
        <Input
          ref={inputRef}
          onChange={onChange}
          rightSection={
            <CloseButton
              aria-label={clearButtonAriaLabel}
              onClick={handleSearchClear}
              size="lg"
              variant="white"
            />
          }
          size="lg"
          value={value}
        />
      </form>
    </Box>
  );
}
