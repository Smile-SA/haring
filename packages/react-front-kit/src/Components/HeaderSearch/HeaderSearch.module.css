.search {
  bottom: 0;
  left: 0;
  padding: 30px 64px;
  position: absolute;
  right: 0;
  translate: 0 100%;

  /*noinspection CssInvalidAtRule*/
  @mixin light {
    background: var(--mantine-color-white);
  }
  /*noinspection CssInvalidAtRule*/
  @mixin dark {
    background: var(--mantine-color-black);
  }
}
