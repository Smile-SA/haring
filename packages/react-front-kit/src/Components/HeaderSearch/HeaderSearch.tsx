'use client';

import type { ISearchBarProps } from '../SearchBar/SearchBar';
import type { BoxProps } from '@mantine/core';
import type { ReactElement } from 'react';

import { Box } from '@mantine/core';

import { SearchBar } from '../SearchBar/SearchBar';

import classes from './HeaderSearch.module.css';

export interface IHeaderSearchProps extends ISearchBarProps {
  boxProps?: BoxProps;
}

/** Additional props will be forwarded to the [Mantine Box component](https://mantine.dev/core/box) */
export function HeaderSearch(props: IHeaderSearchProps): ReactElement {
  const { boxProps, ...searchBarProps } = props;

  return (
    <Box className={classes.search} {...boxProps}>
      <SearchBar {...searchBarProps} />
    </Box>
  );
}
