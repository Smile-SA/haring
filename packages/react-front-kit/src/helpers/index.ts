export * from './nestedObject';
export * from './typeGuard';
