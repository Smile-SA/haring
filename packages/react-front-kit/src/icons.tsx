import type { Icon } from '@phosphor-icons/react';

import * as Icons from '@phosphor-icons/react';

const iconsToIgnore = ['IconBase', 'IconContext'];
const iconEntries = Object.entries(Icons).filter(
  ([key]) => !iconsToIgnore.includes(key)
) as [string, Icon][];

export const icons = Object.fromEntries(iconEntries);

export const iconsElements = Object.fromEntries(
  iconEntries.map(([key, Icon]) => [key, <Icon key={key} />])
);
