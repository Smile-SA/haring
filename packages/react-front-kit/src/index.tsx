/* eslint-disable react-refresh/only-export-components */
'use client';

// component exports
export type { ICardHeaderProps } from './Components/CardHeader/CardHeader';
export { CardHeader } from './Components/CardHeader/CardHeader';
export type { IMenuLanguagesProps } from './Components/MenuLanguages/MenuLanguages';
export { MenuLanguages } from './Components/MenuLanguages/MenuLanguages';
export type { IButtonListOrDropdownProps } from './Components/ButtonListOrDropdown/ButtonListOrDropdown';
export { ButtonListOrDropdown } from './Components/ButtonListOrDropdown/ButtonListOrDropdown';
export { ActionBar } from './Components/ActionBar/ActionBar';
export type { IActionBarProps } from './Components/ActionBar/ActionBar';
export { ActionRowOverflow } from './Components/ActionRowOverflow/ActionRowOverflow';
export type {
  IActionRowOverflowProps,
  IActionRowOverflowAction,
} from './Components/ActionRowOverflow/ActionRowOverflow';
export { BitConverter } from './Components/BitConverter/BitConverter';
export type { IBitConverterProps } from './Components/BitConverter/BitConverter';
export { Breadcrumbs } from './Components/Breadcrumbs/Breadcrumbs';
export type { IBreadcrumbsProps } from './Components/Breadcrumbs/Breadcrumbs';
export { CollapseButtonControlled } from './Components/CollapseButton/CollapseButtonControlled';
export type { ICollapseButtonControlledProps } from './Components/CollapseButton/CollapseButtonControlled';
export { CollapseButton } from './Components/CollapseButton/CollapseButton';
export type { ICollapseButtonProps } from './Components/CollapseButton/CollapseButton';
export { ConfirmModal } from './Components/ConfirmModal/ConfirmModal';
export type { IConfirmModalProps } from './Components/ConfirmModal/ConfirmModal';
export { DocumentBox } from './Components/DocumentBox/DocumentBox';
export type { IDocumentBoxProps } from './Components/DocumentBox/DocumentBox';
export { DocumentList } from './Components/DocumentList/DocumentList';
export type {
  IDocument,
  IDocumentListProps,
} from './Components/DocumentList/DocumentList';
export { DropdownButton } from './Components/DropdownButton/DropdownButton';
export type { IDropdownButtonProps } from './Components/DropdownButton/DropdownButton';
export { FilterList } from './Components/FilterList/FilterList';
export type { IFilterListProps } from './Components/FilterList/FilterList';
export { Header } from './Components/Header/Header';
export type { IHeaderProps } from './Components/Header/Header';
export { HeaderMobile } from './Components/HeaderMobile/HeaderMobile';
export type { IHeaderMobileProps } from './Components/HeaderMobile/HeaderMobile';
export { HeaderNav } from './Components/HeaderNav/HeaderNav';
export type {
  IHeaderNavMenu,
  IHeaderNavProps,
} from './Components/HeaderNav/HeaderNav';
export { HeaderSearch } from './Components/HeaderSearch/HeaderSearch';
export type { IHeaderSearchProps } from './Components/HeaderSearch/HeaderSearch';
export { InfoCard } from './Components/InfoCard/InfoCard';
export type {
  IContentItem,
  IInfoCardProps,
} from './Components/InfoCard/InfoCard';
export { Motif } from './Components/InfoCard/Motif';
export type { IMotifProps } from './Components/InfoCard/Motif';
export { Pagination } from './Components/Pagination/Pagination';
export type { IPaginationProps } from './Components/Pagination/Pagination';
export { Preview } from './Components/Preview/Preview';
export type { IPreviewProps } from './Components/Preview/Preview';
export { ResponsiveTabs } from './Components/ResponsiveTabs/ResponsiveTabs';
export type { IResponsiveTabs } from './Components/ResponsiveTabs/ResponsiveTabs';
export { SearchableCheckboxList } from './Components/SearchableCheckboxList/SearchableCheckboxList';
export type {
  ISearchableCheckboxListProps,
  ICheckbox,
} from './Components/SearchableCheckboxList/SearchableCheckboxList';
export { SearchBar } from './Components/SearchBar/SearchBar';
export type { ISearchBarProps } from './Components/SearchBar/SearchBar';
export { SelectableList } from './Components/SelectableList/SelectableList';
export type { ISelectableListProps } from './Components/SelectableList/SelectableList';
export { SidebarFilters } from './Components/SidebarFilters/SidebarFilters';
export type {
  ISidebarFiltersProps,
  ISidebarFilter,
} from './Components/SidebarFilters/SidebarFilters';
export {
  getRecursiveMenu,
  SidebarMenu,
} from './Components/SidebarMenu/SidebarMenu';
export type {
  IMenuItem,
  ISidebarMenuProps,
} from './Components/SidebarMenu/SidebarMenu';
export { SummaryBox } from './Components/SummaryBox/SummaryBox';
export type { ISummaryBoxProps } from './Components/SummaryBox/SummaryBox';
export { SwitchableView } from './Components/SwitchableView/SwitchableView';
export type {
  IDataView,
  ISwitchableViewProps,
} from './Components/SwitchableView/SwitchableView';
export { Thumbnail } from './Components/Thumbnail/Thumbnail';
export type { IThumbnailProps } from './Components/Thumbnail/Thumbnail';
export { ThumbnailGrid } from './Components/ThumbnailGrid/ThumbnailGrid';
export type { IThumbnailGridProps } from './Components/ThumbnailGrid/ThumbnailGrid';
// layout exports
export { FoldableColumnLayout } from './Layouts/FoldableColumnLayout/FoldableColumnLayout';
export type { IFoldableColumnLayoutProps } from './Layouts/FoldableColumnLayout/FoldableColumnLayout';
// helper exports
export {
  addPathAndDepth,
  flattenNestedObjects,
  getClosestItemFromPath,
  setChildrenToTree,
} from './helpers';
// type exports
export type { IThumbnail, IThumbnailAction, IThumbnailData } from './types';
