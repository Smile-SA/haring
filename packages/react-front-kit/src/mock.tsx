/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
