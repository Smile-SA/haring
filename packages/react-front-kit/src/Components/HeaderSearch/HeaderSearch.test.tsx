import { renderWithProviders } from '@smile/react-front-kit-shared/src/test-utils';

import { HeaderSearch } from './HeaderSearch';

describe('Header', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(<HeaderSearch />);
    expect(container).toMatchSnapshot();
  });
});
