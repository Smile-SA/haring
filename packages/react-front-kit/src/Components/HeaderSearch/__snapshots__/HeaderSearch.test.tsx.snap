// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearch matches snapshot 1`] = `
<div>
  <div
    class="mantine-3ozie3"
  >
    <div
      class="mantine-Input-wrapper mantine-1v7s5f8"
    >
      <div
        aria-invalid="false"
        class="mantine-Input-input mantine-1dmg58i"
      >
        <form
          class="mantine-whzeux"
        >
          <div
            class="mantine-InputWrapper-root mantine-TextInput-root mantine-12ka740"
          >
            <div
              class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-8qx5cx"
            >
              <input
                aria-invalid="false"
                aria-label="Search"
                class="mantine-Input-input mantine-TextInput-input mantine-oxqjuq"
                id="header-search-test"
                type="text"
              />
              <div
                class="mantine-1nmvwx4 mantine-Input-rightSection mantine-TextInput-rightSection"
              >
                <button
                  aria-label="Clear"
                  class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-CloseButton-root mantine-ozrqr9"
                  type="button"
                >
                  <svg
                    fill="none"
                    style="width: 1.75rem; height: 1.75rem;"
                    viewBox="0 0 15 15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
